client/src/components/browseRecipes/Category.svelte

```diff
@@ -0,0 +1,94 @@
+<script>
+  import { createEventDispatcher, onMount } from 'svelte';
+  import { categoriesApi } from '@/api';
+  const dispatch = createEventDispatcher();
+
+  function handleChange({ target }) {
+    dispatch('categoryToggle', {
+      value: target.value,
+      text: target.options[target.selectedIndex].text,
+    });
+  }
+
+  let categoriesList = [];
+  let isExpanded;
+
+  onMount(async () => {
+    isExpanded = false;
+    categoriesList = await categoriesApi.fetchAll();
+  });
+
+  function expand() {
+    isExpanded = !isExpanded;
+  }
+</script>
+
+<main>
+  <button class="expand" on:click={expand}>Filteri</button>
+  <div class="categoryContainer {isExpanded && 'extend'}">
+    <div class="text">Filtriraj po kategoriji:</div>
+    <form>
+      <select id="categorySelect" name="categorySelect" on:change={handleChange}>
```

Comment:
so this is how we do selects in svelte https://svelte.dev/tutorial/select-bindings

basically it would look like this in your use case
```js
let value = null;
function handleChange() {
  dispatch('categoryToggle', { value });
}

<select id="categorySelect" name="categorySelect" bind:value on:change={handleChange}>
  <option value="0">Not selected</option>
  {#each categoriesList as { id, name }}
    <option value={id}>{name}</option>
  {/each}
</select>
```

Reply:
also, do you need this id and name anymore?

Reply:
also also, you can do something like this `dispatch{..., { value: +value }` and now you don't need to cast value into int every time you use it